Agorex-io/BackEnd
# Add token un/verified endpoints

It's bad practice, but since @JonathonDunford  and  I are currently the only two coding on the project and I need to move forward with backend endpoints, I created a branch from my unmerged PR. 

This branch adds an endpoint for submitting new tokens from the Frontend. It's tested and works well.

Tokens are stored in the database, but also in a JSON. This may be a temporary approach, as we could just store tokens in the database. 

The ability to add unverified tokens from the frontend will open up the pathway to request that a token be verified alongside information entered on the frontend. This would remove the need for Agorex-io/tokenbase all together. We could include or link a tutorial / requirements from the frontend and have a user fill out the form for unverified tokens.

There is a placeholder select on the frontend for this functionality where a user can check "Request verification". It is currently disabled.

In addition to adding new tokens, I added endpoints for fetching all of the unverified/verified/both tickers combined with their token information, which greatly improves speed on the frontend. 

If you'd like to test this out with the frontend:
1. clone this branch
2. run Docker following the Backend readme
3. clone the new frontend
4. Open the token dropdown. Try to select "verified" "unverified" or "all" from the drop-down to see how the frontend queries the new endpoints to get ticker and token information quickly. 
5. Try to add any token that pops up in from the [rinkeby etherscan ERC20 search](https://rinkeby.etherscan.io/token-search).
6. Add a brief description and submit token.
